com.unity.render-pipelines.high-definition/Runtime/Material/GGXConvolution/IBLFilterGGX.cs

```diff
@@ -153,9 +184,165 @@ public override void Cleanup()
+                int numTilesXHRHalf = (texWidthHalf + (tileSize - 1)) / tileSize;
+                int numTilesYHRHalf = (texHeightHalf + (tileSize - 1)) / tileSize;
+                cmd.DispatchCompute(m_PlanarReflectionFilteringCS, m_PlanarReflectionDownScaleKernel, numTilesXHRHalf, numTilesYHRHalf, 1);
+
+                // Given that mip to mip in compute doesn't work, we have to do this :(
+                cmd.CopyTexture(m_PlanarReflectionFilterTex1, 0, 0, 0, 0, texWidthHalf, texHeightHalf, m_PlanarReflectionFilterTex0, 0, currentMipSource + 1, 0, 0);
+                cmd.CopyTexture(m_PlanarReflectionFilterDepthTex1, 0, 0, 0, 0, texWidthHalf, texHeightHalf, m_PlanarReflectionFilterDepthTex0, 0, currentMipSource + 1, 0, 0);
+
+                // Update the parameters for the next mip
+                currentTexWidth = currentTexWidth >> 1;
+                currentTexHeight = currentTexHeight >> 1;
+                texWidthHalf = texWidthHalf >> 1;
+                texHeightHalf = texHeightHalf >> 1;
+                currentMipSource++;
+            }
+        }
+
+        override public void FilterPlanarTexture(CommandBuffer cmd, RenderTexture source, ref PlanarTextureFilteringParameters planarTextureFilteringParameters, RenderTexture target)
+        {
+            // First we need to make sure that our intermediate textures are the big enough to do our process (these textures are squares)
+            CheckIntermediateTexturesSize(source.width, source.height);
+
+            // Then we need to build a mip chain (one for color, one for depth) that we will sample later on in the process
+            BuildColorAndDepthMipChain(cmd, source, planarTextureFilteringParameters.captureCameraDepthBuffer, ref planarTextureFilteringParameters);
+
+            // Init the mip descent
+            int texWidth = source.width;
+            int texHeight = source.height;
+
+            // First we need to copy the Mip0 (that matches perfectly smooth surface), no processing to be done on it
+            cmd.CopyTexture(m_PlanarReflectionFilterTex0, 0, 0, 0, 0, texWidth, texHeight, target, 0, 0, 0, 0);
+
+            // Initialize the parameters for the descent
+            int mipIndex = 1;
+            int tileSize = 8;
+            // The number of mips that are used at the the end is fixed
+            const float roughnessStep = 1.0f / k_GgxIblMipCountMinusOne;
+            // Based on the initial texture resolution, the number of available mips for us to read from is variable and is based on the maximal texture width
+            int numMipsChain = (int)(Mathf.Log((float)texWidth, 2.0f) - 1.0f);
+            float currentRoughness = roughnessStep;
+            float rtScaleFactor = texWidth / (float)m_PlanarReflectionFilterTex0.rt.width;
+            texWidth = texWidth >> 1;
+            texHeight = texHeight >> 1;
+
+            // Loop until we have an 8x8 texture
+            while (texWidth >= 8 && texHeight >= 8)
```

Comment:
the loop should be k_GgxIblMipCountMinusOne
We should reach roughness of 1 at mip 6 and stop there.

Reply:
ok, fixing that